homeassistant/components/kodi/browse_media.py

```diff
@@ -161,55 +155,94 @@ def item_payload(item, media_library):
 
     Used by async_browse_media.
     """
+    title = item["label"]
+
+    thumbnail = item.get("thumbnail")
+    if thumbnail:
+        thumbnail = media_library.thumbnail_url(thumbnail)
+
     if "songid" in item:
-        media_content_type = MEDIA_TYPE_TRACK
-        media_content_id = f"{item['songid']}"
+        return BrowseMedia(
```

Comment:
Instead of creating the instance and returning inside each if/elif block, set local variables and just return the BrowseMedia instance once at the end.

Reply:
That was actually an idea from Paulus. I can roll that back, no problem. But this way it is more clear what's playable and expandable IMO.

Reply:
Every if/elif/else block will set those as local variables so I don't think there's a difference in clarity. We will just wait with instantiating the BrowseMedia and returning it to the end.

Reply:
I moved it back at the end.